### PR TITLE
Add a configuration option for the executable path

### LINKED ISSUE #19: Add a config option to set path to executable
Rather than simply expecting `rzk` to be in the `PATH`, allow users to specify a custom path using [`configuration`](https://code.visualstudio.com/api/references/contribution-points#contributes.configuration) settings

### PR DESCRIPTION
**Pull request by aabounegm**
Resolves #19

### CODE REVIEW

**Comment by fizruk** [comment]
Great, thank you! I think proper documentation is needed. How do I configure this locally? Do I **need** to configure it?

**Comment by fizruk** [comment]
By the way, I think it would be good to have CI check `npm i && npm run compile` as well as `vsce package`?

**Comment by aabounegm** [comment]
> How do I configure this locally? Do I **need** to configure it?

You do not need to configure it. It defaults to `null`, which means to just use `rzk` (and expect it to be on the `PATH`). If you want to test with a particular version during development, for example, just go to the settings and set the path:

image

If you meant "locally" to one specific project, just override it in `.vscode/settings.json` in that project

> By the way, I think it would be good to have CI check `npm i && npm run compile` as well as `vsce package`

Sounds like a good idea, but should it be part of this PR?

**Comment by fizruk** [comment]
> Sounds like a good idea, but should it be part of this PR?

Yes, please, extend README with the documentation about configuration options (we have one for now, but perhaps we will have more in the future).

File: README.md [hunk 1]
```diff
@@ -15,3 +15,13 @@ See [Changelog](CHANGELOG.md) for recent updated and changes.
...
+
+## Configuration
+
+Extension settings can be configured by going to the settings page (using the menu `File > Preferences > Settings`, or using the shortcut <kbd>CTRL + ,</kbd>).
+
+The currently available settings are:
+
+| Name       | Type             | Default value | Description |
+| ---------- | ---------------- | ------------- | ----------- |
+| `rzk.path` | `string \| null` | `null`        | The path to the `rzk` executable to use for the language server. Defaults to using the globally available one on the `PATH` |
```
Comment (fizruk) [inline README.md]:
```suggestion
| `rzk.path` | `string \| null` | `null`        | The path to the `rzk` executable to use for the language server. `null` (default) means that `rzk` executable available in `PATH` will be used. |
```

File: README.md [hunk 2]
```diff
@@ -15,3 +15,13 @@ See [Changelog](CHANGELOG.md) for recent updated and changes.
 More examples:
 
 ![Syntax highlighting example.](images/example-relfunext2.png)
+
+## Configuration
+
+Extension settings can be configured by going to the settings page (using the menu `File > Preferences > Settings`, or using the shortcut <kbd>CTRL + ,</kbd>).
```
Comment (fizruk) [inline README.md]:
```suggestion
Extension settings can be configured by going to the settings page (using the menu `File > Preferences > Settings`, or using the shortcut <kbd>CTRL + ,</kbd> or <kbd>⌘ + ,</kbd>).
```

**Review by fizruk — APPROVED** [review]

**Pull request merged**